pkg/apis/workflow/v1alpha1/validation_utils.go

```diff
@@ -0,0 +1,118 @@
+package v1alpha1
```

Comment:
Is this file still needed?

Reply:
Yes, `validateNoCycles` and `validateWorkflowFieldNames` are needed by `ContainerSetTemplate.Validate`. I put them in `validation_utils.go` since other validation may need them later